src/Common/src/Abstractions/HealthChecks/IAsyncHealthContributor.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Threading.Tasks;
+
+namespace Steeltoe.Common.HealthChecks
+{
+    /// <summary>
+    /// Implement this interface and add to DI to be included in health checks
+    /// </summary>
+    public interface IAsyncHealthContributor
+    {
+        /// <summary>
+        /// Gets an identifier for the type of check being performed
+        /// </summary>
+        string Id { get; }
+
+        /// <summary>
+        /// Check the health of a resource
+        /// </summary>
+        /// <returns>The result of checking the health of a resource</returns>
+        Task<HealthCheckResult> CheckHealthAsync();
```

Comment:
Not a big deal, but I was just wondering if the "Async" on the name "CheckHealthAsync" is necessary?  If you think it is meaningful, leave it, but might be unnecessary and could just be called "CheckHealth".

Reply:
I was just mirroring the Microsoft [HealthCheck](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.diagnostics.healthchecks.healthcheckservice.checkhealthasync?view=dotnet-plat-ext-3.1#Microsoft_Extensions_Diagnostics_HealthChecks_HealthCheckService_CheckHealthAsync_System_Threading_CancellationToken_) naming. I am OK with any name.